### ISSUE TITLE
[BUG] Unhandled exception ... Editor is already disposed

### ISSUE THREAD

**Issue by walles** [DataDog/datadog-for-intellij-platform#37]
## Describe the bug

IntellJ sometimes reports this in its IDE Internal Errors dialog.

```
Unhandled exception in [ComponentManager(ProjectImpl@329748819), ClientId(value=Host), CoroutineName(com.datadog.intellij.ucap.editor.CodeInsightEditorInlayManager), Dispatchers.Default]

com.intellij.openapi.util.TraceableDisposable$DisposalException: Editor is already disposed
	at com.intellij.openapi.util.TraceableDisposable.throwDisposalError(TraceableDisposable.java:48)
	at com.intellij.openapi.editor.impl.EditorImpl.throwDisposalError(EditorImpl.java:1102)
	at com.intellij.openapi.editor.impl.view.LogicalPositionCache.checkDisposed(LogicalPositionCache.java:191)
	at com.intellij.openapi.editor.impl.view.LogicalPositionCache.getLineInfo(LogicalPositionCache.java:176)
	at com.intellij.openapi.editor.impl.view.LogicalPositionCache.logicalPositionToOffset(LogicalPositionCache.java:111)
	at com.intellij.openapi.editor.impl.view.EditorCoordinateMapper.logicalPositionToOffset(EditorCoordinateMapper.java:95)
	at com.intellij.openapi.editor.impl.view.EditorCoordinateMapper.logicalToVisualPosition(EditorCoordinateMapper.java:106)
	at com.intellij.openapi.editor.impl.view.EditorCoordinateMapper.offsetToVisualPosition(EditorCoordinateMapper.java:234)
	at com.intellij.openapi.editor.impl.view.EditorView.offsetToVisualPosition(EditorView.java:201)
	at com.intellij.openapi.editor.impl.EditorImpl.lambda$offsetToVisualPosition$34(EditorImpl.java:1555)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runReadAction(AnyThreadWriteThreadingSupport.kt:272)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runReadAction(AnyThreadWriteThreadingSupport.kt:262)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:863)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:66)
	at com.intellij.openapi.editor.impl.EditorImpl.offsetToVisualPosition(EditorImpl.java:1555)
	at com.intellij.openapi.editor.impl.EditorImpl.offsetToVisualPosition(EditorImpl.java:1550)
	at com.intellij.openapi.editor.impl.view.VisualLineFragmentsIterator.<init>(VisualLineFragmentsIterator.java:92)
	at com.intellij.openapi.editor.impl.view.VisualLineFragmentsIterator.lambda$create$0(VisualLineFragmentsIterator.java:31)
	at com.intellij.openapi.editor.impl.view.EditorCoordinateMapper.xyToVisualPosition(EditorCoordinateMapper.java:333)
	at com.intellij.openapi.editor.impl.view.EditorView.xyToVisualPosition(EditorView.java:229)
	at com.intellij.openapi.editor.impl.EditorImpl.lambda$xyToVisualPosition$30(EditorImpl.java:1527)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runReadAction(AnyThreadWriteThreadingSupport.kt:272)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runReadAction(AnyThreadWriteThreadingSupport.kt:262)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:863)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:66)
	at com.intellij.openapi.editor.impl.EditorImpl.xyToVisualPosition(EditorImpl.java:1527)
	at com.intellij.openapi.editor.impl.EditorImpl.lambda$xyToLogicalPosition$38(EditorImpl.java:1599)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runReadAction(AnyThreadWriteThreadingSupport.kt:272)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runReadAction(AnyThreadWriteThreadingSupport.kt:262)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:863)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:66)
	at com.intellij.openapi.editor.impl.EditorImpl.xyToLogicalPosition(EditorImpl.java:1597)
	at com.intellij.openapi.editor.ex.util.EditorScrollingPositionKeeper.savePosition(EditorScrollingPositionKeeper.java:39)
	at com.datadog.intellij.ucap.editor.DatadogInlayEditorDecorator.applyInlays(CodeInsightEditorInlayManager.kt:431)
	at com.datadog.intellij.ucap.editor.DatadogInlayEditorDecorator.access$applyInlays(CodeInsightEditorInlayManager.kt:338)
	at com.datadog.intellij.ucap.editor.DatadogInlayEditorDecorator$tryRemapAndAddInlays$3.invokeSuspend(CodeInsightEditorInlayManager.kt:375)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at com.intellij.openapi.application.impl.DispatchedRunnable.run(DispatchedRunnable.kt:44)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:236)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:25)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:218)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:117)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:43)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:781)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:728)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:750)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:675)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:573)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$18$lambda$17$lambda$16$lambda$15(IdeEventQueue.kt:355)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:857)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$18$lambda$17$lambda$16(IdeEventQueue.kt:354)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$2$lambda$1(IdeEventQueue.kt:1045)
	at com.intellij.openapi.application.WriteIntentReadAction.lambda$run$0(WriteIntentReadAction.java:24)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runWriteIntentReadAction(AnyThreadWriteThreadingSupport.kt:128)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteIntentReadAction(ApplicationImpl.java:916)
	at com.intellij.openapi.application.WriteIntentReadAction.compute(WriteIntentReadAction.java:55)
	at com.intellij.openapi.application.WriteIntentReadAction.run(WriteIntentReadAction.java:23)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$2(IdeEventQueue.kt:1045)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$3(IdeEventQueue.kt:1054)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:109)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:1054)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$18(IdeEventQueue.kt:349)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:395)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [ComponentManager(ProjectImpl@329748819), ClientId(value=Host), CoroutineName(com.datadog.intellij.ucap.editor.CodeInsightEditorInlayManager), StandaloneCoroutine{Cancelled}@ea94996, Dispatchers.Default]
```

## Steps To Reproduce
Unclear, this happens from time to time.

## Screenshots
![Image](https://github.com/user-attachments/assets/b41e7902-a2b2-4ff3-a5fb-207de02c222b)

### Expected Behavior
No exceptions reported by IntelliJ.

## What versions of software are you using?
- Operating System: macOS Sequioia 15.1.1
- IDE version: IntelliJ IDEA 2024.3 (Ultimate Edition) Build #IU-243.21565.193, built on November 13, 2024 
- Datadog plugin version: 1.6.2

**Comment by jfree** [comment]
Thanks for the report Johan, we really appreciate it!  We should have a fix for the next release (v1.7.0).

**Comment by jfree** [comment]
The fix is on the marketplace now.  I'll close this ticket but don't hesitate to re-open it or create a new ticket if the problem occurs again.
